cli/azd/cmd/login.go

```diff
@@ -35,6 +39,18 @@ func (lf *loginFlags) Bind(local *pflag.FlagSet, global *internal.GlobalCommandO
 		false,
 		"When true, log in by using a device code instead of a browser.",
 	)
+	local.StringVar(&lf.clientID, "client-id", "", "The client id for the service principal to authenticate with.")
+	local.StringVar(
+		&lf.clientSecret,
+		"client-secret",
+		"",
+		"The client secret for the service principal to authenticate with.")
```

Comment:
Not sure about the guidance around secrets... Should this secret be passed in via stdin?

Reply:
We now support reading this from stdin for such cases.  We also allow it to come in on the command line.  You opt into having it read from stdin by explicitly setting the value to "", as in --client-secret "".  In that case, we prompt.